File: checks/ruby/lib/section508/checks/images_of_text.rb
# frozen_string_literal: true

module Section508
  module Checks
    # Words that are the content should be real text.
    class ImagesOfText
      TEXT_SRC = /headline|banner-text|sale-title|heading-img|text-banner|hours-sign|promo-text|word-art|slogan|title-banner/i
      LOGO = /logo|wordmark|seal|brand-mark/i

      def self.call(root)
        return [] unless root&.respond_to?(:css)

        Dom.elements(root, "img").each do |image|
          src = image["src"].to_s
          alt = image["alt"].to_s
          next if LOGO.match?(src) || LOGO.match?(alt)
          return ["These words are an image of text, not real text."] if TEXT_SRC.match?(src)
        end
        []
      end
    end
  end
end

File: checks/python/src/section508/checks/images_of_text.py
import re

from section508.dom import attr, select

TEXT_SRC = re.compile(
    r"headline|banner-text|sale-title|heading-img|text-banner|hours-sign|promo-text|word-art|slogan|title-banner",
    re.I,
)
LOGO = re.compile(r"logo|wordmark|seal|brand-mark", re.I)


def check(root):
    if not root:
        return []

    for image in select(root, "img"):
        src = attr(image, "src") or ""
        alt = attr(image, "alt") or ""
        if LOGO.search(src) or LOGO.search(alt):
            continue
        if TEXT_SRC.search(src):
            return ["These words are an image of text, not real text."]
    return []
